lib/staging/tls/tls.cpp

```diff
@@ -692,6 +697,36 @@ std::uint32_t ServerConnection::m_count{0};
 std::mutex ServerConnection::m_cv_mutex;
 std::condition_variable ServerConnection::m_cv;
 
+namespace {
+
+std::unique_ptr<std::filesystem::path> keylog_file;
+std::unique_ptr<TlsKeyLoggingServer> keylog_server;
+
+void keylog_callback(const SSL* ssl, const char* line) {
+    std::string key_log_msg = "TLS Handshake keys: " + std::string(line);
+    log_info(key_log_msg);
+
+    if (keylog_server and keylog_server->get_fd() != -1) {
```

Comment:
SSL_get_ex_data could be used here to get the correct keylog_server and keylog_file
The callback could also then handle the case where keylog_server etc. are not available.